c++/src/kj/async-coroutine-test.c++

```diff
@@ -381,9 +381,9 @@ KJ_TEST("co_await only sees coroutine destruction exceptions if promise was not
       awaitPromise(kj::mv(rejectedThrowyDtorPromise)).wait(waitScope));
 }
 
-#if (!_MSC_VER || defined(__clang__)) && !__aarch64__
+#if !_MSC_VER && !__aarch64__
 // TODO(msvc): This test relies on GetFunctorStartAddress, which is not supported on MSVC currently,
-//   so skip the test.
+//   so skip the test. It also appears to fail on clang-cl, which uses the MSVC backend.
```

Comment:
It looks like the relevant code in `async-inl.h` (in `PtmfHelper`) is guarded by `#if __GNUG__`, which evaluates false on clang-cl, which would indeed break this test.

The code in question depends on the details of the Itanium C++ ABI, which I'm guessing means it wouldn't work on clang-cl, which uses the MSVC ABI. So it seems the check is correct and skipping the test is the right thing to do.

Comment might be better as:

```suggestion
//   so skip the test. Note this is an ABI issue, so clang-cl is also not supported.
```